WordPress/gutenberg
# [RNMobile] Enable reusable block only in WP.com sites

`gutenberg-mobile` PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3490
`WordPress-iOS` PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/16475
`WordPress-Android` PR: https://github.com/wordpress-mobile/WordPress-Android/pull/14623





## Description

This PR adds a capability into the editor to limit the availability of the reusable block to WP.com sites, until we manage to address [the issue related to fetching reusable blocks in self-hosted sites](https://github.com/wordpress-mobile/gutenberg-mobile/issues/3457).

When the new capability `reusableBlock` is `false`, we do the following actions:
1. Unregister `core/block` (reusable block), this way we prevent the reusable block data to be fetched in case it's already in the post ([code reference](https://github.com/WordPress/gutenberg/pull/31744/files#diff-634bfd5cd865bf9a1e36bfbac1ed489aeb740a2ff5a46e29174b41b14972d77cR63))
2. Prevent fetching reusable blocks when the editor is mounted ([code reference](https://github.com/WordPress/gutenberg/pull/31744/files#diff-44d02c0638cdd23c2fdc5fd67fbc806d176985b5e6d50388d5628d6fc9544446R20))

## How has this been tested?



We have to test that the reusable block is enabled in WP.com sites but not on self-hosted:

<details>
<summary><strong>WordPress.com site - Reusable block is enabled</strong></summary>

Create a WordPress.com site or use an already created one.
- Create a post in the web version
- Add some blocks and create a Reusable block
- Open the app with metro running (Gutenberg-mobile)
- Add the WordPress.com site
- Edit the previous created post
- Check that the Reusable block is shown
- Tap on it and check that the title of the Reusable blocks is shown

</details>

<details>
<summary><strong>WordPress.com site (Atomic) - Reusable block is enabled</strong></summary>

Create an Atomic site via WordPress.com site or use an already created one.
- Create a post in the web version
- Add some blocks and create a Reusable block
- Open the app with metro running (Gutenberg-mobile)
- Add the WordPress.com site
- Edit the previous created post
- Check that the Reusable block is shown
- Tap on it and check that the title of the Reusable blocks is shown

</details>

<details>
<summary><strong>Self-hosted site with Jetpack - Reusable block is disabled</strong></summary>

Create a self-hosted site with Jetpack (I used https://jurassic.ninja/create/) or use an already created one.
- Create a post in the web version
- Add some blocks and create a Reusable block
- Open the app with metro running (Gutenberg-mobile)
- Add the self-hosted site
- Edit the previous created post
- Check that the Reusable block is displayed as unsupported

</details>

<details>
<summary><strong>Self-hosted site without Jetpack - Reusable block is disabled</strong></summary>

Create a self-hosted site without Jetpack (I created a local one using [this instructions](https://developer.wordpress.org/block-editor/contributors/develop/getting-started/#local-environment)) or use your own.
- Create a post in the web version
- Add some blocks and create a Reusable block
- Open the app with metro running (Gutenberg-mobile)
- Add the self-hosted site
- Edit the previous created post
- Check that the Reusable block is displayed as unsupported

</details>

## Screenshots 

WP.com site|Self-hosted site
--|--
<image src=https://user-images.githubusercontent.com/14905380/117959761-9d08fc00-b31c-11eb-97f9-d68e39e2059a.png width=500>|<image src=https://user-images.githubusercontent.com/14905380/117959771-a003ec80-b31c-11eb-890d-441d496839e5.png width=500>


## Types of changes




Bug fix

## Checklist:
